### PR TITLE
Trigger grammar-used hooks no matter how text editors are created

### PR DESCRIPTION
**Pull request by maxbrunsfeld**
Fixes https://github.com/atom/atom/issues/12333

### CODE REVIEW

File: src/workspace.coffee [hunk 1]
```diff
@@ -160,6 +161,13 @@ class Workspace extends Model
       @activeItemSubscriptions.add(titleSubscription) if titleSubscription?
       @activeItemSubscriptions.add(modifiedSubscription) if modifiedSubscription?
 
+  subscribeToAddedItems: ->
+    @onDidAddPaneItem ({item, pane, index}) =>
+      if item instanceof TextEditor
+        grammarSubscription = item.observeGrammar(@handleGrammarUsed.bind(this))
+        item.onDidDestroy -> grammarSubscription.dispose()
+        @emitter.emit 'did-add-text-editor', {textEditor: item, pane, index}
+
```
Comment (maxbrunsfeld) [inline src/workspace.coffee]: It seems like we need to look at every editor being added to the workspace in order to make sure we always trigger the `grammar-used` hooks.

File: src/workspace.coffee [hunk 2]
```diff
@@ -383,8 +391,7 @@ class Workspace extends Model
   #
   # Returns a {Disposable} on which `.dispose()` can be called to unsubscribe.
   onDidAddTextEditor: (callback) ->
-    @onDidAddPaneItem ({item, pane, index}) ->
-      callback({textEditor: item, pane, index}) if item instanceof TextEditor
+    @emitter.on 'did-add-text-editor', callback
```
Comment (maxbrunsfeld) [inline src/workspace.coffee]: I figured we should avoid a duplicate `instanceof TextEditor` check, so I moved the emission of the `did-add-text-editor` event into the callback above.

**Comment by nathansobo** [comment]
This looks good to me. Basically there are ways editors can be added to the workspace without going through `openTextFile`. Thanks for fixing this Max.

**Pull request merged**